i-VRESSE/workflow-builder
# Dont make checkbox required always

Fixes #67

To test 
1. 1st way repeat steps in #67 
    1. Goto haddock3-download app at https://app.netlify.com/sites/wonderful-noether-53a9e8/deploys/62d930fd6466c80009103c7d
    2. Instead of using netlify hosted app you can also run locally from root of repo run `yarn && yarn dev`  and goto http://localhost:3000     
    3.  Add `mdref` node to workflow
    5. Expand molecule subform
    6. Press `+` button, this will add an unchecked checklbox
    7. Submit form
    8. Switch to text mode of workflow to see `mol_shape_1 = false`, before this PR it would have an invalid form.
1. 2nd way with story book
    1. from root of repo run `yarn storybook` 
    2. Goto http://localhost:6007
    4. In left sidebar expand `Checkbox`
    5. Select `Unchecked Checkbox In Array Is Submiteable` story
    6. Press `Submit` button, should not complain and in actions tab have a new entry with onSubmit[0].formData key have value `[false, true]`. The story presses the submit button when opening the story, so you will see 2 actions.
3. 3rd way with unit test
    1. Change to  `packages/form/` and run `yarn test`
    2. It should run `packages/form/src/IvresseCheckBoxWidget.test.tsx`and pass.

~This PR also adds vitest to form package and upgrades rjsf to latest version.~ Moved to #100
